### PR TITLE
[BEAM-22] Add WindowedValue#explodeWindows

### PR DESCRIPTION
**Pull request by tgroh**
Be sure to do all of the following to help us incorporate your contribution
quickly and easily:

 - [x] Make sure the PR title is formatted like:
   `[BEAM-<Jira issue #>] Description of pull request`
 - [x] Make sure tests pass via `mvn clean verify`. (Even better, enable
       Travis-CI on your fork and ensure the whole test matrix passes).
 - [x] Replace `<Jira issue #>` in the title with the actual Jira issue
       number, if there is one.
 - [x] If this contribution is large, please file an Apache
       [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.txt).

---

This takes an existing WindowedValue and returns a Collection of
WindowedValues, each of which is in exactly one window.

Use the explode implementation on DoFnRunnerBase

### CODE REVIEW

**Comment by tgroh** [comment]
R: @bjchambers

**Comment by bjchambers** [comment]
Update description and commit message to reflect the new name.

File: sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java [hunk 1]
```diff
@@ -175,6 +176,18 @@ public T getValue() {
   public abstract Collection<? extends BoundedWindow> getWindows();
 
   /**
+   * Returns a collection of {@link WindowedValue WindowedValues} identical to this one, except each
+   * is in exactly one of the windows that this {@link WindowedValue} is in.
+   */
+  public Collection<WindowedValue<T>> explodeWindows() {
```
Comment (bjchambers) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java]: Any reason not to use an Iterable and/or Iterables.transform?

Comment (tgroh) [inline sdks/java/core/src/main/java/org/apache/beam/sdk/util/WindowedValue.java]: I can use an Iterable. Done

I think using Iterables.transform is less readable than the imperative style.

**Comment by tgroh** [comment]
Jenkins Failure is [BEAM-208](https://issues.apache.org/jira/browse/BEAM-208)

**Comment by bjchambers** [comment]
LGTM

**Pull request closed without merging**